### DISCUSSION TITLE
umi4，怎么使用手动注册乾坤，而不是在app.ts上注册

### DISCUSSION THREAD

**Discussion by zxt-gitDemo** [umijs/umi#9082] [Q&A]
app.ts是在项目初始化的时候执行，但是这个时候我还没获取到配置，无法写入注册，只能在组件内部进行注册，但是umi4文档中好像没有提供这种写法，我要怎么办

**Comment by fz6m** [comment]
获取到远程配置前不渲染项目。

**Reply by kelisiWu123** [reply]
获取到远程配置前不渲染项目。您好，请问一下，这个应该如何做到呢

**Reply by fz6m** [reply]
See https://github.com/umijs/umi/issues/12267#issuecomment-2047425572
